account_product_fiscal_classification/models/product_template.py

```diff
@@ -75,15 +77,58 @@ def _update_vals_fiscal_classification(self, vals):
                 }
             )
         elif vals.get("supplier_taxes_id") or vals.get("taxes_id"):
-            raise ValidationError(
-                _(
-                    "You can not create or write products with"
-                    " 'Customer Taxes' or 'Supplier Taxes'\n."
-                    "Please, use instead the 'Fiscal Classification' field."
-                )
-            )
+            self._find_or_create_classification(vals)
         return vals
 
     @api.constrains("categ_id", "fiscal_classification_id")
     def _check_rules_fiscal_classification(self):
         self.env["account.product.fiscal.rule"].check_product_templates_integrity(self)
+
+    def _find_or_create_classification(self, vals):
+        """Find the correct Fiscal classification,
+        depending of the taxes, or create a new one, if no one are found."""
+        # search for matching classication
+        domain = []
+        purchase_tax_ids = vals.get("supplier_taxes_id")
+        sale_tax_ids = vals.get("taxes_id")
+        for elm in ("supplier_taxes_id", "taxes_id"):
+            if elm in vals:
+                del vals[elm]
+        taxe_ids = []
+        if sale_tax_ids:
+            if isinstance(sale_tax_ids, int):
+                sale_tax_ids = [sale_tax_ids]
+            domain.append(("sale_tax_ids", "in", sale_tax_ids))
+            taxe_ids.extend(sale_tax_ids)
+        if purchase_tax_ids:
+            if isinstance(purchase_tax_ids, int):
+                purchase_tax_ids = [purchase_tax_ids]
+            domain.append(("purchase_tax_ids", "in", purchase_tax_ids))
+            taxe_ids.extend(purchase_tax_ids)
+        classification = self.env["account.product.fiscal.classification"].search(
+            domain
+        )
+        if not classification:
+            # Create a dedicate classification for these taxes combination
+            classif_vals = {
```

Comment:
This part could be set in the account.product.fiscal.classification model, in a function named : 
_prepare_classification_from_vals(self, purchase_tax_ids, sale_tax_ids)